packages/libs/routing/src/auth.ts

```diff
@@ -1,17 +1,17 @@
-import { CloudFrontResultResponse } from "aws-lambda";
+import { Header, Response } from "./types";
 
 export function getUnauthenticatedResponse(
-  authorizationHeader: string | null,
+  authorizationHeaders: Header[] | null,
   authentication: { username: string; password: string } | undefined
-): CloudFrontResultResponse | null {
+): Response | null {
   if (authentication && authentication.username && authentication.password) {
     const validAuth =
       "Basic " +
       Buffer.from(
         authentication.username + ":" + authentication.password
       ).toString("base64");
 
-    if (authorizationHeader !== validAuth) {
+    if (!authorizationHeaders || authorizationHeaders[0]?.value !== validAuth) {
       return {
```

Comment:
For this, this was a quick way to return a response directly from origin request handler in CloudFront. But as I mentioned, it might be better returned as Node.js req/res instead to be more standard and converted with the lambda@edge compatibility code (converting Node.js req/res -> CloudFront req/res)

Reply:
You mean using IncomingMessage and ServerResponse?

That would pretty much mean implementing a state machine that manipulates a pair of them passed from the lambda-at-edge (or other) implementation? I was thinking this could be just a nice and easily testable library of pure functions. 🙂 Returning a new ServerResponse or IncomingMessage would get annoying fast (typing, streaming), I think.

Then again, moving as much logic to shared code as possible would, of course, make other implementations simpler. I'll play around with it a bit when I have time. But if you have strong opinions about how this should work, please share.

Reply:
Yeah, that's what I meant. I think for now it might be ok to keep it simpler first and do the conversion as needed later. Also Cloudflare workers that doesn't support Node.js yet so anyway they might need to be shimmed for Cloudflare workers? (they are on a restricted V8 environment)

Reply:
Apparently Cloudflare has Node.js support now: https://blog.cloudflare.com/node-js-support-cloudflare-workers/ The current 1 MB limit is a problem for Next.js though. And I'm not sure what their time limits are.

Anyway, I changed the approach slightly, returning "Route" objects instead, which lambda-at-edge handles. The idea is that eventually the only logic in lambda-at-edge request handler would be going through the different cases and either returning a redirect, rendering and returning a page or making an S3 request.

Reply:
Yeah, that makes sense. The 1MB limit may be ok for small projects but for most, even mine which is medium-sized, each page is like 950 kB compressed. So to fit all pages into one bundle would be hard. Although static pages-only apps can be supported